### PR TITLE
Clarify help text of kubectl replace

### PR DESCRIPTION
**Pull request by j3ffml**
Help the next poor soul avoid the confusion I had (#12352) when

```
$ kubectl create -f pod.yaml
$ kubectl replace -f pod.yaml
```

doesn't work.

cc @caesarxuchao

### CODE REVIEW

File: pkg/kubectl/cmd/replace.go [hunk 1]
```diff
@@ -34,7 +34,8 @@ import (
 const (
 	replace_long = `Replace a resource by filename or stdin.
 
-JSON and YAML formats are accepted.
+If replacing an existing resource, the complete resource spec must be provided
+(this can be obtained by 'kubectl get <kind> -o yaml'). JSON and YAML formats are accepted.
```
Comment (caesarxuchao) [inline pkg/kubectl/cmd/replace.go]: maybe change to <kind/name>?

File: docs/user-guide/kubectl/kubectl_replace.md [hunk 2]
```diff
@@ -40,7 +40,8 @@ Replace a resource by filename or stdin.
 
 Replace a resource by filename or stdin.
 
-JSON and YAML formats are accepted.
+If replacing an existing resource, the complete resource spec must be provided
+(this can be obtained by 'kubectl get <kind> -o yaml'). JSON and YAML formats are accepted.
```
Comment (bgrant0607) [inline docs/user-guide/kubectl/kubectl_replace.md]: Thanks. I'd change the examples, too.
```
kubectl get pod mypod -o yaml | sed 's/image: [^:]*:.*$/v4/' | kubectl replace -f -
```

Comment (bgrant0607) [inline docs/user-guide/kubectl/kubectl_replace.md]: Another try:
```
kubectl get pod mypod -o yaml | sed 's/\(image: [^:]*\):.*$/\1:v4/' | kubectl replace -f -
```

File: pkg/kubectl/cmd/replace.go (hunk 1 above)
Comment (bgrant0607) [inline pkg/kubectl/cmd/replace.go]: Should use whatever syntax we use in help messages, actually.

Comment (jlowdermilk) [inline pkg/kubectl/cmd/replace.go]: `TYPE NAME` is the syntax other help messages use.

**Comment by bgrant0607** [comment]
LGTM

**Pull request merged**